examples/stress_tests/many_components.rs

```diff
@@ -124,23 +127,51 @@ fn stress_test(num_entities: u32, num_components: u32, num_systems: u32) {
     // spawn a bunch of entities
     for _ in 1..=num_entities {
         let num_components = rng.gen_range(1..10);
-        let components = component_ids.choose_multiple(&mut rng, num_components);
+        let components: Vec<ComponentId> = component_ids
+            .choose_multiple(&mut rng, num_components)
+            .copied()
+            .collect();
 
         let mut entity = world.spawn_empty();
-        for &component_id in components {
-            let value: u8 = rng.gen_range(0..255);
-            OwningPtr::make(value, |ptr| {
+        // We use `ManuallyDrop` here as we need to avoid dropping the u8's when `values` is dropped
+        // since ownership of the values is passed to the world in `insert_by_ids`. 
+        // But we do want to deallocate the memory when values is dropped.
+        let mut values: Vec<ManuallyDrop<u8>> = components
+            .iter()
+            .map(|_id| ManuallyDrop::new(rng.gen_range(0..255)))
+            .collect();
+        let ptrs: Vec<OwningPtr> = values
+            .iter_mut()
+            .map(|value| {
+                // ManuallyDrop is repr(transparent) so casting it to a *mut u8 is valid.
+                let ptr = value as *mut ManuallyDrop<u8> as *mut u8;
+                // SAFETY: value is non null since it was initialized from an &mut above
                 #[allow(unsafe_code)]
+                let ptr = unsafe { NonNull::new_unchecked(ptr) };
                 // SAFETY:
-                // component_id is from the same world
-                // value is u8, so ptr is a valid reference for component_id
+                // * values is valid until the end of the for block and this pointer is consumed by `insert_by_ids`.
+                // * ptr was created from a u8, so is aligned, and has correct provenance.
+                // * ptr was created from an exclusive reference and nothing else reads or writes to `values`.
+                #[allow(unsafe_code)]
                 unsafe {
-                    entity.insert_by_id(component_id, ptr);
+                    OwningPtr::new(ptr)
                 }
-            });
+            })
+            .collect();
+        // SAFETY:
+        // * component_id's are from the same world
+        // * `values` was initialized above, so references are valid
+        #[allow(unsafe_code)]
+        unsafe {
+            entity.insert_by_ids(&components, ptrs.into_iter());
         }
     }
 
+    println!(
```

Comment:
Should this be `info!` so it can be configured with the logging framework?

Reply:
I'm not sure. I see some examples use `info` and some use `println`.  This example was using println already, so I just kept it consistent.